src/libs/actions/Session/Reauthenticator.ts

```diff
@@ -0,0 +1,77 @@
+import Onyx from 'react-native-onyx';
+import {reauthenticate} from '@libs/Authentication';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type Session from '@src/types/onyx/Session';
+
+let isOffline = false;
+let active = false;
+let currentActiveSession: Session = {};
+let timer: NodeJS.Timeout;
+// the delay before requesting a reauthentication once activated
+// we think in that timing a "natural" reauthentication could happen (a session expired in the carousel is the only exception)
+// and we wish not to overlap and make a double reauthentication
```

Comment:
What do you think if we update to

```suggestion
// When the session is expired we will give it this time to reauthenticate via normal flows, like the Reauthentication middleware, in an attempt to not duplicate authentication requests
// also, this is an arbitrary number so we may tweak as needed
```